### PR TITLE
Update Deserialization

### PR DESCRIPTION
**Pull request by twicki**
## Technical Description

Update the deserializer to set the ID generator to the right place that Passes do not fail.

### Resolves / Enhances

Broken passes after deserialisation

### CODE REVIEW

File: dawn/src/dawn/Serialization/IIRSerializer.cpp [hunk 1]
```diff
@@ -152,6 +154,12 @@ static void computeInitialDerivedInfo(const std::shared_ptr<iir::StencilInstanti
   }
 }
 
+static void checkAndUpdateMaxID(int newID, int& maxID) {
```
Comment (jdahm) [inline dawn/src/dawn/Serialization/IIRSerializer.cpp]: Do we need a function here? Is this just
```
#include <cmath>
#include <algorithm>
maxID = std::max(maxID, std::abs(newID));
```

Comment (twicki) [reply dawn/src/dawn/Serialization/IIRSerializer.cpp]: do you want to have it as that one liner everywhere or do you want it wrapped in a function anyway?

**Comment by jdahm** [comment]
Other than the question above, this looks good!

**Comment by twicki** [comment]
launch jenkins

**Pull request merged**